c/src/eval_ctx.c

```diff
@@ -63,7 +82,12 @@ plcs_errors plcs_eval_ctx_set_str_eval_param(plcs_string_evaluators ix, const ch
     return PLCS_EIX_OVERFLOW;
   }
 
-  ctx.string_evaluators[ix].value = value;
+  char *maybe_dup = plcs_strdup(value);
+  if (maybe_dup == NULL) {
+    return PLCS_EALLOCATION;
+  }
+
+  ctx.string_evaluators[ix].value = maybe_dup;
```

Comment:
Should we free the previous value if it's not null?

Reply:
indeed. Good catch. Thanks